tmpl/blog/introducing-solo5.md

```diff
@@ -0,0 +1,125 @@
+I'm excited to announce the release of
+[Solo5](https://github.com/djwillia/solo5/tree/mirage) via mirage.io!
+Solo5 is essentially a kernel library that bootstraps the hardware and
+forms a base (similar to Mini-OS) from which unikernels can be built.
+It runs on fully virtualized x86 hardware (e.g., KVM/QEMU), using
+`virtio` device interfaces.
+
+Importantly, Solo5 is integrated (to some extent) with the Mirage
+toolstack, so the Solo5 version of the Mirage toolstack can build
+Mirage unikernels that run directly on KVM/QEMU instead of Xen.  As
+such, Solo5 can be considered an alternative to Mini-OS in the Mirage
+stack.  [Try it out
+today!](https://github.com/djwillia/solo5/tree/mirage)
+
+In the rest of this post, I'll give a bit of motivation about why I
+think the lowest layer of the unikernel is interesting and important,
+as well as a rough overview of the steps I took to create Solo5.
+Thanks for reading!
+
+### Why focus so far down the software stack?
+
+When people think about Mirage unikernels, one of the first things
+that comes to mind is the use of a high-level language (OCaml).
+Indeed, the Mirage community has invested lots of time and effort
+producing implementations of traditional system components (e.g., an
+entire TCP stack) in OCaml.  The pervasive use of OCaml contributes to
```

Comment:
TCP stack -> [TCP stack](https://github.com/mirage/mirage-tcpip)
i.e. linking to the repo